src/backend/utils/indexer.js

```diff
@@ -7,7 +7,58 @@ const { client } = require('../lib/elastic');
 const { logger } = require('./logger');
 
 const index = 'posts';
-const type = 'post';
+
+// creates 'posts' index and add settings and mapping for autocomplete
+const setupPostsIndex = async () => {
+  try {
+    const response = await client.indices.exists({ index });
+    if (response.statusCode === 404) {
+      await client.indices.create({
```

Comment:
Given that you're the only person who understands this code, and we'll have to maintain it, can you add a big comment above this index creation explaining how it works, linking to relevant docs, etc?

Reply:
[The documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-edgengram-tokenizer.html#analysis-edgengram-tokenizer) and [this video](https://www.youtube.com/watch?v=FSPwonNOldk&ab_channel=Codetuber)  helped me understand